core/pubnub_coreapi.c

```diff
@@ -281,22 +281,141 @@ enum pubnub_res pubnub_set_state(pubnub_t*   pb,
                                  char const* state)
 {
     enum pubnub_res rslt;
-
     PUBNUB_ASSERT(pb_valid_ctx_ptr(pb));
 
     pubnub_mutex_lock(pb->monitor);
     if (!pbnc_can_start_transaction(pb)) {
         pubnub_mutex_unlock(pb->monitor);
         return PNR_IN_PROGRESS;
     }
-
     rslt = pbcc_set_state_prep(
         &pb->core, channel, channel_group, uuid ? uuid : pbcc_uuid_get(&pb->core), state);
     if (PNR_STARTED == rslt) {
         pb->trans            = PBTT_SET_STATE;
         pb->core.last_result = PNR_STARTED;
         pbnc_fsm(pb);
         rslt = pb->core.last_result;
+        if (rslt == PNR_STARTED) {
+            int ch_cnt = 0;
+            int cg_cnt = 0;
+            int tot_ch = 0;
+            int tot_cg = 0;
+            if (channel){
+                tot_ch = 1;
+                for (int i=0; i < strlen(channel); i++) { tot_ch = (channel[i] == ',') ? tot_ch + 1 : tot_ch; }
+            }
+            if (channel_group){
+                tot_cg = 1;
+                for (int i=0; i < strlen(channel_group); i++) { tot_cg = (channel_group[i] == ',') ? tot_cg + 1 : tot_cg; }
+            }
+            
+            int buff_size = ((tot_ch + tot_cg) * strlen(state)) + (channel ? strlen(channel) : 1) + (channel_group ? strlen(channel_group) : 1) + 20;
+            char * json_state = (char*)malloc(buff_size);
+            char * core_state;
+            if (pb->core.state != NULL && buff_size != sizeof(pb->core.state)){
+                core_state = (char*)realloc((char*)pb->core.state, buff_size);
+            }
+            else if (pb->core.state == NULL){
+                core_state = (char*)malloc(buff_size);
+            }
+            int m_len = 0;
+            if (json_state != NULL && core_state != NULL){
+                pb->core.state = core_state;
+                m_len = strlen("{");
```

Comment:
I'd call it maybe `cursor`. Or maybe `json_cursor`? Or maybe you can explain what actually `m_len` stands for or specifically `m` cause I can figure out `len` on my own ;)

Reply:
`m` is memory and `len` is for length. So `m_len` is for memory length.

Reply:
ok..may be change `m` to `mem` ?